src/d3d9/d3d9_device.cpp

```diff
@@ -3980,7 +3980,18 @@ namespace dxvk {
       Flags &= ~D3DLOCK_DISCARD;
 
     if (!(Flags & D3DLOCK_NO_DIRTY_UPDATE) && !(Flags & D3DLOCK_READONLY)) {
+      if (pBox && MipLevel != 0) {
+        D3DBOX scaledBox = *pBox;
+        scaledBox.Left   <<= MipLevel;
+        scaledBox.Right  <<= MipLevel;
+        scaledBox.Top    <<= MipLevel;
+        scaledBox.Bottom <<= MipLevel;
+        scaledBox.Back   <<= MipLevel;
+        scaledBox.Front  <<= MipLevel;
```

Comment:
Do we need to clamp this? Eg. if there are several mips with a 1xY size. That 1 could get scaled up outside the bounds of the texture.

Reply:
Any idea what to clamp this to?

Reply:
Dont need to clamp the left/top/front because it's all unsigned.